### PR TITLE
Normalize Line Endings

### PR DESCRIPTION
**Pull request by kjeremy**
Line endings with git always confuse me so let's be explicit.

In doing this I ran `git add --renormalize .` on my repo per https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings and it converted a bunch of files. I then backed it all out and ran it again and only this one `Cargo.toml` file changed so I suspect line endings are just weird.

### CODE REVIEW

**Comment by lnicola** [comment]
See also `* text=auto !eol`.

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +1,9 @@
+*.el text eof=LF
+*.js text eof=LF
+*.json text eof=LF
+*.ron text eof=LF
+*.rs text eof=LF
+*.toml text eof=LF
+*.ts text eof=LF
 crates/ra_syntax/test_data/** -text eof=LF
 crates/ra_ide_api/src/snapshots/** -text eof=LF
```
Comment (matklad) [inline .gitattributes]: Let's drop this line, I think we no longer commit snapshots

**Comment by matklad** [comment]
@lnicola do you mean there's a single line we can add to apply to all files?

File: .gitattributes [hunk 2]
```diff
@@ -1,2 +1,8 @@
+*.el text eof=LF
+*.js text eof=LF
+*.json text eof=LF
+*.ron text eof=LF
+*.rs text eof=LF
+*.toml text eof=LF
+*.ts text eof=LF
```
Comment (matklad) [inline .gitattributes]: I also wonder if it effectively undos the undoing of `line_ending = unix` in rustfmt.toml? Should we add it back?

**Comment by lnicola** [comment]
That's what I took from https://www.reddit.com/r/git/comments/7tfh7c/comment/dtcw509.

**Comment by kjeremy** [comment]
I just tried `* text=auto !eol` and it added `^M` which is what we don't want.

**Comment by kjeremy** [comment]
Rust repo uses `* text=auto eol=lf` which I think is what we want.

**Comment by kjeremy** [comment]
Ok force pushed and should be good. `cargo xtask format` and `cargo xtask codegen` with modified files have the correct line endings so I don't think we need to change the rustfmt configuration.

**Pull request merged**